ast/src/parsed/display.rs

```diff
@@ -574,14 +658,8 @@ impl<Ref: Display> Display for Expression<Ref> {
             Expression::Tuple(items) => write!(f, "({})", format_list(items)),
             Expression::LambdaExpression(lambda) => write!(f, "{}", lambda),
             Expression::ArrayLiteral(array) => write!(f, "{array}"),
-            Expression::BinaryOperation(left, op, right) => write!(f, "({left} {op} {right})"),
-            Expression::UnaryOperation(op, exp) => {
-                if op.is_prefix() {
-                    write!(f, "{op}{exp}")
-                } else {
-                    write!(f, "{exp}{op}")
-                }
-            }
+            Expression::BinaryOperation(_, _, _) => self.format_binary_operation(f),
+            Expression::UnaryOperation(_, _) => self.format_unary_operation(f),
             Expression::IndexAccess(index_access) => write!(f, "{index_access}"),
```

Comment:
Since we don't put any parentheses at the very outside of a group of binary operations, I fear we also have to include IndexAccess and FunctionCall now. Because `(a + b)(2)` will now be formatted as `a + b(2)` and `(a + b)[2]` also as `a + b[2]`.

Reply:
I will test